extension/src/background/messageListener/freighterApiMessageListener.ts

```
@@ -130,6 +140,78 @@ export const freighterApiMessageListener = (
     }
   };
 
+  const submitToken = async () => {
+    try {
+      const { contractId, networkPassphrase: reqNetworkPassphrase } =
+        request as ExternalRequestToken;
+
+      const networkPassphrase =
```

Seems like defaulting to mainnet when there is no `networkPassphrase` is an assumption likely to fail. What about failing early if we have no network passphrase here? I think even though the type for `ExternalRequestToken` is a bit loose, in practice we should always have a network passphrase right?

---

This comment definitely makes sense @aristidesstaffieri , but we do actually tend to default to MAINNET in the case of `no network passphrase` in our other API methods (like signTransaction). This is actually a model that a lot of wallets in the ecosystem follow, as well. So, probably makes sense to keep this